File: _old/Client/admin-panel/src/entities/orders/model/order.ts
import { User } from "@entities/users";
import { Product } from "@entities/products";

export enum OrderStatus {
	Created,
	PendingForPayment,
	Paid,
	InTransit,
	Delivered,
	Cancelled
}

export type Order = {
	id: number;
	createdAt: string;
	updatedAt: string;
	status: OrderStatus;
	shipAddress: string;
	orderInformation: string;
	customer: User;
	products?: Product[] | null;
};

File: _old/_oldServer/Server/Entities/Order.cs
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;

using Server.Shared.Interfaces;
using Server.Shared.Enums;

namespace Server.Entities
{
    [Table("Orders")]
    public class Order : IIdentifiable
    {
        [Key]
        [Required]
        public int Id { get; set; }

        [Required]
        public int UserId { get; set; }

        [Required]
        public DateTime CreatedAt { get; set; } = new DateTime();

        [Required]
        public DateTime UpdatedAt { get; set; } = new DateTime();

        [Required]
        public OrderStatus Status { get; set; } = OrderStatus.Created;

        [Required]
        [StringLength(80)]
        public string ShipAddress { get; set; } = string.Empty;

        [Required]
        [MaxLength(400)]
        public string OrderInformation { get; set; } = string.Empty;

        public User? User { get; set; }

        public ICollection<ProductOrderLink>? OrderProducts { get; set; }
    }
}
